File: 백준/Silver/1755. 숫자놀이/숫자놀이.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

struct numToStr{
    char str[100];
    int num;
} numToStr;

int compare(const void* a, const void* b) {
    const struct numToStr* str1 = (const struct numToStr*)a;
    const struct numToStr* str2 = (const struct numToStr*)b;

    return strcmp(str1->str, str2->str);
}

int main()
{
    int m, n;

    scanf("%d %d", &m, &n);

    char * numDic[10] = { "zero" ,"one" ,"two" ,"three" ,"four" ,"five" ,"six" ,"seven" ,"eight" ,"nine" };

    struct numToStr* numToStr = malloc(sizeof(struct numToStr) * (n - m + 1));

    for (int i = m; i <= n; i++) {
        if (i >= 10) {
            char temp[100];
            strcpy(temp, numDic[i / 10]);
            strcat(temp, numDic[i % 10]);
            strcpy(numToStr[i - m].str, temp);
            numToStr[i - m].num = i;
        }
        else {
            char temp[100];
            strcpy(temp, numDic[i % 10]);
            strcpy(numToStr[i - m].str, temp);
            numToStr[i - m].num = i;
        }
    }

    qsort(numToStr, n - m + 1, sizeof(struct numToStr), compare);

    for (int i = 0; i <= n - m; i++) {
        printf("%d ", numToStr[i].num);

        if (i % 10 == 9) {
            printf("\n");
        }
    }

    free(numToStr);

    return 0;
}

File: 백준/Silver/1755. 숫자놀이/숫자놀이.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m, n;

        String numDic[] = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };

        m = sc.nextInt();
        n = sc.nextInt();

        List<numToStr> numToStrArr = new ArrayList<>();

        for (int i = m; i <= n; i++) {
            if (i >= 10) {
                String tmepStr = numDic[i / 10] + " " + numDic[i % 10];
                numToStrArr.add(new numToStr(tmepStr, i));
            } else {
                String tmepStr = numDic[i % 10];
                numToStrArr.add(new numToStr(tmepStr, i));
            }
        }

        // 오름차순 정렬
        Collections.sort(numToStrArr);

        for (int i = 0; i < numToStrArr.size(); i++) {
            System.out.print(numToStrArr.get(i).num + " ");

            if (i % 10 == 9) {
                System.out.print("\n");
            }
        }

        sc.close();
    }
}

// 2가지 데이터를 저장할 클래스
class numToStr implements Comparable<numToStr> {
    public String str;
    public int num;

    // 객체를 정렬하는 것은
    // https://www.daleseo.com/java-comparable-comparator/ 참고

    public numToStr(String tmepStr, int i) {
        this.str = tmepStr;
        this.num = i;
    }

    public int compareTo(numToStr n2s) {
        return this.str.compareTo(n2s.str);
    }
}